pkg/client/group_mgmt_client.go

```diff
@@ -60,7 +64,53 @@ type Argument struct {
 	JobId string `json:"job_id,omitempty"`
 }
 
-func newGroupMgmtClient(ipAddress, username, password, apiVersion string, waitOnJobs bool) *GroupMgmtClient {
+type ClientOption func(*GroupMgmtClient)
```

Comment:
Let's think about where these definitions belong.  If the NsGroupService is the consumer entry point, we'll want to locate these functions near it, probably in group_service_factory.go.  Maybe they are really ServiceOptions rather than ClientOptions.  (Honestly I find the client/service distinctions confusing.)